frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/common-template-validations/common-templates-validations.ts

```diff
@@ -0,0 +1,82 @@
+import { getFieldId } from '../../../utils/vm-settings-tab-utils';
+import { iGetIn } from '../../../../../utils/immutable';
+import { UpdateOptions } from '../../types';
+import { iGetVmSettingAttribute } from '../../../selectors/immutable/vm-settings';
+import { iGetLoadedCommonData } from '../../../selectors/immutable/selectors';
+import { VMSettingsField, VMWizardProps, VMSettingsRenderableField } from '../../../types';
+import {
+  TEMPLATE_TYPE_LABEL,
+  TEMPLATE_WORKLOAD_LABEL,
+  TEMPLATE_OS_LABEL,
+  TEMPLATE_FLAVOR_LABEL,
+} from '../../../../../constants/vm';
+import { CommonTemplatesValidation } from './validation-types';
+
+const IDToJsonPath = {
+  [getFieldId(VMSettingsField.MEMORY)]: '.spec.domain.resources.requests.memory',
+  [getFieldId(VMSettingsField.CPU)]: '.spec.domain.cpu.cores',
+};
+
+export const getTemplateValidations = (
+  options: UpdateOptions,
+  fieldId: VMSettingsRenderableField,
+): CommonTemplatesValidation[] => {
+  const { getState, id } = options;
+  const state = getState();
+
+  // Get OS, workload-profile and flavor attributes
+  const os = iGetVmSettingAttribute(state, id, VMSettingsField.OPERATING_SYSTEM);
+  const flavor = iGetVmSettingAttribute(state, id, VMSettingsField.FLAVOR);
+  const workloadProfile = iGetVmSettingAttribute(state, id, VMSettingsField.WORKLOAD_PROFILE);
+
+  // Get all the templates from common-templates
+  const commonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
+
+  let relevantTemplates;
+  // Get templates based on OS and WorkloadProfile
+  if (os && workloadProfile) {
+    relevantTemplates = (commonTemplates || []).filter(
+      (template) =>
+        iGetIn(template, ['metadata', 'labels', TEMPLATE_TYPE_LABEL]) === 'base' &&
+        iGetIn(template, ['metadata', 'labels', `${TEMPLATE_OS_LABEL}/${os}`]) &&
+        iGetIn(template, ['metadata', 'labels', `${TEMPLATE_WORKLOAD_LABEL}/${workloadProfile}`]),
+    );
+
+    // Get the exact template when a flavor is chosen
+    if (flavor && flavor !== 'Custom') {
+      relevantTemplates = relevantTemplates.filter((t) =>
+        iGetIn(t, ['metadata', 'labels', `${TEMPLATE_FLAVOR_LABEL}/${flavor}`]),
+      );
+    }
+    const myValidations: CommonTemplatesValidation[][] = [];
+    relevantTemplates.forEach((relevantTemplate) => {
+      myValidations.push(
+        JSON.parse(iGetIn(relevantTemplate, ['metadata', 'annotations', 'validations'])),
+      );
+    });
+
+    // Proceed if there are no validations from common templates
+    if (myValidations.length === 0) {
+      return [];
+    }
+
+    // Proceed if there are no validation for the specific fieldId
+    if (!(fieldId in IDToJsonPath)) {
+      return [];
+    }
+
+    // Get all relevant validations for the specific fieldId
+    const relevantFieldValidations: CommonTemplatesValidation[] = [];
+    myValidations.forEach((validationArray: CommonTemplatesValidation[]) => {
```

Comment:
can you make a "getValidation" method ( similar to [comment about getRelevantTemplates](https://github.com/openshift/console/pull/3839/files#r362150735) ) ?